src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -829,6 +830,102 @@ ov::pass::NopSliceBeforeGatherElements::NopSliceBeforeGatherElements() {
+        if (const auto& begin_constant_node = ov::util::get_constant_from_source(begin_node)) {
+            const auto& values = begin_constant_node->cast_vector<int64_t>();
+            auto begin_mask = strided_slice_node->get_begin_mask();
+            const auto dim_size = begin_mask.size();
+            if (values.size() != dim_size) {
+                return false;
+            }
+            for (size_t i = 0; i < dim_size; ++i) {
+                if (!begin_mask[i] && values[i]) {
+                    return false;
+                }
+            }
+        }
+
+        auto end_node = strided_slice_node->get_input_node_shared_ptr(2);
+        if (const auto& end_constant_node = ov::util::get_constant_from_source(end_node)) {
+            const auto& values = end_constant_node->cast_vector<int64_t>();
+            auto end_mask = strided_slice_node->get_end_mask();
+            const auto dim_size = end_mask.size();
+            if (values.size() != dim_size) {
+                return false;
+            }
+            auto in_shape = strided_slice_node->get_input_partial_shape(0);
+            if (in_shape.size() < dim_size) {
+                return false;
+            }
+            for (size_t i = 0; i < dim_size; ++i) {
+                if (!end_mask[i] && values[i] < static_cast<int64_t>(in_shape[i].get_max_length())) {
+                    return false;
+                }
+            }
+        }
+
+        return replace_output_update_name(strided_slice_node->output(0),
+                                          strided_slice_node->get_input_node_shared_ptr(0)->get_default_output());
+    };
+    auto m = std::make_shared<pattern::Matcher>(pattern, matcher_name);
+    register_matcher(m, matcher_pass_callback);
+}
+
+ov::pass::NopStridedSliceByShape::NopStridedSliceByShape() {
+    MATCHER_SCOPE(NopStridedSliceByShape);
+    auto slice = pattern::wrap_type<op::v8::Slice>();
+
+    auto input = pattern::any_input();
+    auto begin_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto end_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto optional_stride_const = pattern::optional<ov::op::v0::Constant>();
+    auto pattern = pattern::wrap_type<ov::op::v1::StridedSlice>({input, begin_const, end_const, optional_stride_const});
+    ;
+
+    ov::matcher_pass_callback matcher_pass_callback = [=](pattern::Matcher& m) {
+        auto node = m.get_match_root();
+        if (node == nullptr) {
+            return false;
+        }
+        auto strided_slice_node = std::dynamic_pointer_cast<ov::op::v1::StridedSlice>(node);
+        if (!strided_slice_node->get_shrink_axis_mask().empty() || !strided_slice_node->get_new_axis_mask().empty() ||
+            !strided_slice_node->get_ellipsis_mask().empty()) {
+            return false;
+        }
+        if (strided_slice_node->get_input_partial_shape(0) != strided_slice_node->get_output_partial_shape(0)) {
```

Comment:
what if we have dynamic dimensions in input and output shapes then it is not necessarily that we have a deal with Identity operation. For static shapes, the transformations looks valid but for dynamic shapes it can be not true.

Reply:
This transformation is applicable only in case `use_shape_for_elimination`

Reply:
correct note!
use_shape_for_elimination means that we use shapes for elimination, but shapes can be dynamic here
please take a look at eliminate_reshape and eliminate_nop transformations
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/2c55f4ac-f4cd-4092-9928-e21e08fb941c)

Reply:
@please take a look to review. In prev commit I decided to check static shapes

Reply:
could you align this check with "eliminate_reshape and eliminate_nop" transformations?
it is better if this check will be the same for all transformations of this kind.